syntax = "proto3";

package dwc;

// Contador distribuído de palavras
service DistributedWordCounter {
  // Serviço que recebe Lista de palavras e dicionário de frequência 
  // de palavras.
  rpc count (DwcRequest) returns (DwcResponse) {}
}

// Lista de palavras para serem contadas
message DwcRequest {
  // Lista de palavras para serem contadas
  repeated string words = 1;
}

// Dicionário de frequência de palavras
message DwcResponse {
  // Dicionário de frequência de palavras
  map<string, int32> word_frequency = 1;
}
